### PR TITLE
cc-wrapper: Always export environment variables for binutils

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Motivation for this change

Before, this only happened when cross compiling. This is less of a hack (and soon the entire `preWrap` script will go), uniformity is good.

###### Things done

Built Linux and Darwin stdenvs.

----

CC @ali-abrar This narrows down the bugs we've seen to the `infixSalt`-ing, basically.

### CODE REVIEW

File: pkgs/build-support/cc-wrapper/setup-hook.sh [hunk 1]
```diff
@@ -39,9 +37,26 @@ if [ -n "@coreutils_bin@" ]; then
...
+export ${ENV_PREFIX}CXX=@named_cxx@
+
+for CMD in \
+    cpp \
+    ar as nm objcopy ranlib strip strings size ld windres
+do
+    if
+        PATH=$_PATH type -p "@binPrefix@$CMD" > /dev/null
+    then
+        export "${ENV_PREFIX}${CMD^^}=@binPrefix@${CMD}";
```
Comment (Ericson2314) [inline pkgs/build-support/cc-wrapper/setup-hook.sh]: @Dridus Sorry but I changed it back as new we require newish bash anyways.

**Review by shlevy — APPROVED** [review]
Just glanced through, but LGTM

File: pkgs/build-support/cc-wrapper/setup-hook.sh (hunk 1 above)
Comment (Dridus) [inline pkgs/build-support/cc-wrapper/setup-hook.sh]: yeah I saw, no worries

**Comment by Ericson2314** [comment]
Well, this one is uncharacteristically small an readable :).

**Comment by orivej** [comment]
IMHO we went in the wrong direction, we should have stopped (and should stop) exporting AR, AS, LD etc. [1] (everything but CC [2]) for cross rather than start exporting them for native, and provide the cross utilities under unprefixed names in the PATH instead. This would have avoided (and will avoid) much debugging and unnecessary workarounds against upstream build scripts, both for native and for cross packaging [3].

[1] This does not apply to BUILD_{var} and {var}_FOR_BUILD.
[2] CC is used consistently upstream and it smooths the difference between gcc and clang.
[3] For example `ed` will just build, without #32489.

**Comment by Ericson2314** [comment]
@orivej my original goal was to end up always prefixing all tools with their target platforms (wrapper script only for multi-platform tools). I think the current design makes more sense with that objective.

Yours probably would be epitomized with instead prefixing the binaries with `build-` and `target-`, which would also work, and avoid a lot of the of the `add-flags.sh` and `role` craziness. But it would mean that `buildPackages.buidlPackges.my_compiler` (e.g. `buildPackages.stdenv.cc`) would not work.

**Pull request merged**